lib/ansible/modules/windows/win_regedit.ps1

```diff
@@ -21,313 +22,326 @@
+Function Convert-RegExportHexStringToByteArray($string) {
     # Remove 'hex:' from the front of the string if present
-    $String = $String.ToLower() -replace '^hex\:',''
+    $string = $string.ToLower() -replace '^hex\:',''
 
     # Remove whitespace and any other non-hex crud.
-    $String = $String.ToLower() -replace '[^a-f0-9\\,x\-\:]',''
+    $string = $string.ToLower() -replace '[^a-f0-9\\,x\-\:]',''
 
     # Turn commas into colons
-    $String = $String -replace ',',':'
+    $string = $string -replace ',',':'
 
     # Maybe there's nothing left over to convert...
-    if ($String.Length -eq 0) {
+    if ($string.Length -eq 0) {
         return ,@()
     }
 
     # Split string with or without colon delimiters.
-    if ($String.Length -eq 1) {
-        return ,@([System.Convert]::ToByte($String,16))
-    } elseif (($String.Length % 2 -eq 0) -and ($String.IndexOf(":") -eq -1)) {
-        return ,@($String -split '([a-f0-9]{2})' | foreach-object { if ($_) {[System.Convert]::ToByte($_,16)}})
-    } elseif ($String.IndexOf(":") -ne -1) {
-        return ,@($String -split ':+' | foreach-object {[System.Convert]::ToByte($_,16)})
+    if ($string.Length -eq 1) {
+        return ,@([System.Convert]::ToByte($string,16))
+    } elseif (($string.Length % 2 -eq 0) -and ($string.IndexOf(":") -eq -1)) {
+        return ,@($string -split '([a-f0-9]{2})' | foreach-object { if ($_) {[System.Convert]::ToByte($_,16)}})
+    } elseif ($string.IndexOf(":") -ne -1) {
+        return ,@($string -split ':+' | foreach-object {[System.Convert]::ToByte($_,16)})
     } else {
         return ,@()
     }
 }
 
-# Create the required PSDrives if missing
-if (-not (Test-Path HKCR:\)) {
-    New-PSDrive -Name HKCR -PSProvider Registry -Root HKEY_CLASSES_ROOT
-}
-if (-not (Test-Path HKU:\)) {
-    New-PSDrive -Name HKU -PSProvider Registry -Root HKEY_USERS
-}
-if (-not (Test-Path HKCC:\)) {
-    New-PSDrive -Name HKCC -PSProvider Registry -Root HKEY_CURRENT_CONFIG
+Function Get-ValueType($type) {
+    switch ($type) {
+        "none" { [Microsoft.Win32.RegistryValueKind]::None }
+        "binary" { [Microsoft.Win32.RegistryValueKind]::Binary }
+        "dword" { [Microsoft.Win32.RegistryValueKind]::DWord }
+        "expandstring" { [Microsoft.Win32.RegistryValueKind]::ExpandString }
+        "multistring" { [Microsoft.Win32.RegistryValueKind]::MultiString }
+        "string" { [Microsoft.Win32.RegistryValueKind]::String }
+        "qword" { [Microsoft.Win32.RegistryValueKind]::QWord }
+    }
 }
 
-
-# Convert HEX string to binary if type binary
-if ($type -eq "binary" -and $data -ne $null -and $data -is [string]) {
-    $data = Convert-RegExportHexStringToByteArray($data)
+Function Test-RegistryProperty($path, $name) {
+    # will validate if the registry key contains the property, returns true
+    # if the property exists and false if the property does not
+    try {
+        $value = (Get-Item -Path $path).GetValue($name)
+        if ($value -eq $null) {
```

Comment:
I can't do that here, `$value -ne $null` doesn't return a boolean when we are dealing with an empty multistring value.